### PR TITLE
Add secure cipher suite list to extra args for kubernetes components

### PR DESCRIPTION
**Pull request by vivek-koppuru**
*Issue #, if available:*

*Description of changes:*

Currently our main kubernetes components use some cipher suites that are not considered secure as per the following [list](https://github.com/golang/go/blob/master/src/crypto/tls/cipher_suites.go#L81). Since the components default to certain ones on the list, we are overriding the default by passing them in as a flag. In the future if we need to support more ciphers, we can either add more to the function or have it configurable by the user through the cluster spec.

The following changes for external etcd will come in a followup PR.

Also in this PR:
- include `make release-manifest` code for podIAMConfig
- fix docker unit tests to default testing external etcd and add a test for stacked etcd
- fix spacing issues in audit policy
- refactor tlsvalidator into `validator.go` to make `tls.go` a more generic tls related file

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/657#" title="Author self-approved">vivek-koppuru</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [vivek-koppuru]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by vivek-koppuru** [comment]
Running some tests right now

/hold

**Comment by abhay-krishna** [comment]
@vivek-koppuru: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | 4c53aeac1e2c17211ab5b4f4e10495a2cdae74e7 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/657/eks-anywhere-e2e-presubmit/1461107907352334336) | `/test eks-anywhere-e2e-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=657). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:vivek-koppuru).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: pkg/clusterapi/extraargs.go
Comment (mrajashree) [inline pkg/clusterapi/extraargs.go:61]: nit
```suggestion
func SecureEtcdTlsCipherSuitesExtraArgs() ExtraArgs {
```

**Review by mrajashree — COMMENTED** [review]
Looks great, we might have to set the tls cipher arg for kubelet on worker nodes too?https://github.com/aws/eks-anywhere/blob/main/pkg/providers/vsphere/config/template-md.yaml#L35
https://github.com/aws/eks-anywhere/blob/main/pkg/providers/docker/config/template-md.yaml#L12

**Review by mrajashree — APPROVED** [review]
/lgtm

**Pull request merged**
